File: src/z_lib/api/validate/text.rs
use crate::z_lib::validate::data::ValidateTextError;

pub fn check_text_empty(value: &str) -> Result<(), ValidateTextError> {
    if value.chars().count() == 0 {
        Err(ValidateTextError::Empty)
    } else {
        Ok(())
    }
}
pub fn check_text_too_long(value: &str, max_length: usize) -> Result<(), ValidateTextError> {
    if value.chars().count() > max_length {
        Err(ValidateTextError::TooLong)
    } else {
        Ok(())
    }
}
pub fn check_text_invalid_email(value: &str) -> Result<(), ValidateTextError> {
    if !value.contains("@") {
        Err(ValidateTextError::InvalidEmail)
    } else {
        Ok(())
    }
}

File: src/common/util/validate/text.ts
import { Validator } from "./infra"

import { ValidateResult, ValidateTextError } from "./data"

export function check_text_empty(text: string): ValidateResult<ValidateTextError> {
    if (text.length === 0) {
        return { valid: false, err: { type: "empty" } }
    }
    return { valid: true }
}
export function check_text_tooLong(maxLength: number): Validator<string, ValidateTextError> {
    return (text) => {
        if (text.length > maxLength) {
            return { valid: false, err: { type: "too-long", maxLength } }
        }
        return { valid: true }
    }
}
export function check_text_invalidEmail(text: string): ValidateResult<ValidateTextError> {
    if (text.length === 0) {
        // empty は check_text_empty で判定する
        return { valid: true }
    }
    if (!text.includes("@")) {
        return { valid: false, err: { type: "invalid-email" } }
    }
    return { valid: true }
}
